openedx/core/djangoapps/xblock/runtime/blockstore_runtime.py

```diff
@@ -67,9 +67,19 @@ def get_block(self, usage_id, for_parent=None):
                 # When a former XModule no longer needs to support the old runtime, its parse_xml_new_runtime method
                 # should be removed and its parse_xml() method should be simplified to just call the super().parse_xml()
                 # plus some minor additional lines of code as needed.
-                block = block_class.parse_xml_new_runtime(xml_node, runtime=self, keys=keys)
+                try:
+                    block = block_class.parse_xml_new_runtime(xml_node, runtime=self, keys=keys)
+                except Exception as exc:
+                    message = 'parse_xml_new_runtime xblock error for usage_id {}'.format(usage_id)
+                    log.warning(message)
+                    raise XBlockParseException(message) from exc
             else:
-                block = block_class.parse_xml(xml_node, runtime=self, keys=keys, id_generator=None)
+                try:
+                    block = block_class.parse_xml(xml_node, runtime=self, keys=keys, id_generator=None)
+                except Exception as exc:
+                    message = 'parse_xml xblock error for usage_id {}'.format(usage_id)
+                    log.warning(message)
+                    raise XBlockParseException(message) from exc
```

Comment:
@s0b0lev Can you also please add tests for both types of XBlocks (there are in common/) which verify that the correct exception is thrown in case the OLX is invalid?

Reply:
Added test here: https://github.com/edx/edx-platform/pull/27068/commits/71a3287a13481f360e6a603392ae6ee37ce34535#diff-1aea83268816c6bd0de212b9d353cd8cbeb29a5f0f628352c1a1149690c2b04cR133